scripts/globals/mobs.lua

```diff
@@ -744,3 +744,167 @@ xi.mob.difficulty =
+                }
+            end,
+
+            [xi.job.DRG] = function(x)
+                -- inject "<mob> uses Call Wyvern"
+                actionParams =
+                {
+                    finishCategory = 11,
+                    animationID = 438,
+                    actionID = xi.mobSkill.CALL_WYVERN,
+                    messageID = 101,
+                }
+            end,
+        }
+    end
+
+    params.action = params.action or {}
+    if not actionParams then
+    -- Generic 2-hour animation with no message or options from params table
+        actionParams =
+        {
+            finishCategory = params.action.finishCategory or 11,
+            actionID = params.action.messageID or 307,
+            animationID = params.action.animationID or 439,
+            messageID = params.action.messageID or 0,
+            param = params.action.param or 0,
+        }
+    end
+
+    -- function to execute when pets are actually called (there may be an inactiveTime)
+    local callPetFinish = function(mobArg)
+        if mobArg:isDead() then
+            return
+        end
+
+        -- inject action packet to indicate mob is summoning a pet
+        if not params.noAnimation then
+            if params.inactiveTime > 0 then
+                mobArg:entityAnimationPacket(xi.animationString.CAST_SUMMONER_STOP)
+            else
+                if actionParams then
+                    -- Generic 2-hour animation with no message
+                    mobArg:injectActionPacket(mobArg:getID(), actionParams.finishCategory, actionParams.animationID, 0, 0x18, actionParams.messageID, 0, actionParams.actionID, actionParams.param)
+                end
+            end
+        end
+
+        local spawnPos = mobArg:getSpawnPos()
+        local pos = mobArg:getPos()
+        for _, petId in ipairs(petIds) do
+            local petToSummon = GetMobByID(petId)
+            if
+                petToSummon and
+                not petToSummon:isSpawned()
+            then
+                -- spawn pet around owner
+                petToSummon:setSpawn(pos.x + math.random(-2, 2), pos.y, pos.z + math.random(-2, 2), pos.rot)
+                petToSummon:spawn()
+                -- set home to be the owner's home position
+                petToSummon:setSpawn(spawnPos.x, spawnPos.y, spawnPos.z, spawnPos.rot)
+
+                if params.dieWithOwner then
+                    local listenerName = fmt('OWNER_DEATH_{}', petId)
+                    mobArg:addListener('DEATH', listenerName, function(owner)
+                        local petToKill = GetMobByID(petId)
+                        if petToKill and petToKill:isSpawned() then
+                            petToKill:setHP(0)
+                        end
+
+                        owner:removeListener(listenerName)
+                    end)
+                end
+
+                -- make pet assist with a slight delay to allow spawn to complete so animations don't get bugged
```

Comment:
Mobs don't instantly act on aggro on retail so this probably won't be needed forever, but good idea regardless